src/blueprint/from_config/operators/http.rs

```diff
@@ -30,6 +30,32 @@ impl<'a> MustachePartsValidator<'a> {
   fn new(type_of: &'a config::Type, config: &'a Config, field: &'a FieldDefinition) -> Self {
     Self { type_of, config, field }
   }
+  fn get_nested_type(&self, tail: &[String], mut type_of: &'a config::Type) -> Result<Type, String> {
+    let mut len = tail.len();
+    for item in tail {
+      if let Some(val_type) = get_value_type(type_of, item) {
+        match len {
+          1 => {
+            return Ok(val_type);
+          }
+          _ => {
+            if val_type.is_nullable() {
+              return Err(format!("value '{}' is a nullable type", item.as_str()));
+            }
+          }
+        }
+      } else {
+        return Err(format!("no value '{}' found", item));
+      }
+      let field = type_of.fields.get(item).ok_or(format!("no value '{}' found", item))?;
+      type_of = self
+        .config
+        .find_type(&field.type_of)
+        .ok_or(format!("no value '{}' found", item))?;
+      len -= 1;
+    }
+    unreachable!()
```

Comment:
I think the code could be written without that:
- to resolve the value you need to pass the whole tail array actually, so if we store current resolved value in some mutable variable outside the loop, after the loop is finished we can just return that value
- if while iterating we can't resolve some value just return Err

Reply:
I returned an error instead; imo we can keep the return inside the loop as it is instead of create a variable outside of it